Controller/Api/ContentController.php

```diff
@@ -69,18 +90,74 @@ public function viewAction(Request $request, $id)
     }
 
     /**
-     * Delete
+     * Restore
      *
      * @param Request $request
      * @param integer $id
      *
-     * @return Response
+     * @return JsonResponse
+     */
+    public function restoreAction(Request $request, $id)
+    {
+        $manager = $this->get('opifer.content.content_manager');
+        $repository = $manager->getRepository();
+        $repository->setRetrieveArchived(true);
+        $content = $repository->find($id);
+
+
+        if(is_null($content)) {
+            return new JsonResponse(['success' => false ]);
+        }
+
+        $content->setDeletedAt(null);
+        $em = $this->get('doctrine')->getManager();
+        $em->persist($content);
+        $em->flush();
+        $repository->setRetrieveArchived(false);
+
+        return new JsonResponse(['success' => true ]);
+    }
+
+    /**
+     * Delete
+     *
+     * @param Request $request
+     * @param $id
+     * @return null|JsonResponse|Response
      */
     public function deleteAction(Request $request, $id)
     {
+        return $this->deleteContent($request, $id);
+    }
+
+    /**
+     * Permanent Delete
+     *
+     * @param Request $request
+     * @param $id
+     * @return null|JsonResponse|Response
+     */
+    public function permaAction(Request $request, $id)
+    {
+        return $this->deleteContent($request, $id, true);
+    }
+
+    /**
+     *
+     * @param Request $request
+     * @param $id
+     * @param bool $archive
+     * @return null|JsonResponse|Response
+     */
+    private function deleteContent(Request $request, $id, $archive = false) {
         $manager = $this->get('opifer.content.content_manager');
-        $content = $manager->getRepository()->find($id);
+        $repository = $manager->getRepository();
+        if($archive) { $repository->setRetrieveArchived(true); }
```

Comment:
Could you place the content of the if statement on a new line, and a space after `if`?